File: dotNet/astenn/astenn-core/src/Org/Lestr/Astenn/Configuration/RAMPersistenceDriver.cs
/*
 * Crée par SharpDevelop.
 * Utilisateur: Choupis
 * Date: 30/04/2010
 * Heure: 22:33
 * 
 * Pour changer ce modèle utiliser Outils | Options | Codage | Editer les en-têtes standards.
 */
using System;
using System.Collections.Generic;
using Org.Lestr.Astenn.Plugin;

namespace Org.Lestr.Astenn.Configuration
{
	
	
	public class RAMPersistenceDriver : IPersistenceDriver {


        private Dictionary<string, List<string>> plugins;
	
	
	    public RAMPersistenceDriver()
	    {

            plugins = new Dictionary<string, List<string>>();
	
	    }// END Constructor
	
	
	    public void AddPluginInterface(string pluginInterfaceName) {

            plugins.Add(pluginInterfaceName, new List<string>());
	
	    }// END Method AddPluginInterface
	
	
	    public void RemovePluginInterface(string pluginInterfaceName) {
	
	        plugins.Remove(pluginInterfaceName);
	
	    }// END Method RemovePluginInterface
	
	
	    public bool ExistPluginInterface(string pluginInterfaceName) {
	
	        return plugins.ContainsKey(pluginInterfaceName);
	
	    }// END Method ExistPluginInterface


        public void AddPluginImplementation(string pluginInterfaceName, string pluginImplementationAddress)
        {

            if (!ExistPluginImplementation(pluginInterfaceName, pluginImplementationAddress))
                plugins[pluginInterfaceName].Add(pluginImplementationAddress);
	
	    }// END Method AddPluginImplementation


        public void RemovePluginImplementation(string pluginInterfaceName, string pluginImplementationAddress)
        {

            if (ExistPluginImplementation(pluginInterfaceName, pluginImplementationAddress))
	    	    plugins[pluginInterfaceName].Remove(pluginImplementationAddress);
	
	    }// END Method RemovePluginImplementation


        public bool ExistPluginImplementation(string pluginInterfaceName, string pluginImplementationAddress)
        {
	
	    	return ExistPluginInterface(pluginInterfaceName) && plugins[pluginInterfaceName].Contains(pluginImplementationAddress);
	
	    }// END Method ExistPluginImplementation
	
	
	    public IEnumerable<string> GetPluginInterfacesNames() {
	
	    	return plugins.Keys;
	
	    }// END Method GetPluginImplementationsNames


        public IEnumerable<string> GetPluginImplementationsAddresses(string pluginInterfaceName)
        {
	
	    	return plugins.ContainsKey(pluginInterfaceName) ? plugins[pluginInterfaceName] : new List<string>();
	
	    }// END Method GetPluginImplementationsNames
	    

	}// END Classe RAMPersistenceDriver
	
	
}// END Namespace Astenn.Configuration

File: astenn-core/src/main/java/org/lestr/astenn/configuration/RAMPersistenceDriver.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.lestr.astenn.configuration;

import org.lestr.astenn.plugin.IPersistenceDriver;
import java.util.ArrayList;
import java.util.HashMap;

/**
 *
 * @author PIBONNIN
 */
public class RAMPersistenceDriver implements IPersistenceDriver {


    private HashMap<String, ArrayList<String>> plugins;


    public RAMPersistenceDriver()
    {

        plugins = new HashMap<String, ArrayList<String>>();

    }// END Constructor


    @Override
    public void addPluginInterface(String pluginInterfaceName) {

        plugins.put(pluginInterfaceName, new ArrayList<String>());

    }// END Method addPluginInterface


    @Override
    public void removePluginInterface(String pluginInterfaceName) {

        plugins.remove(pluginInterfaceName);

    }// END Method removePluginInterface


    @Override
    public boolean existPluginInterface(String pluginInterfaceName) {

        return plugins.containsKey(pluginInterfaceName);

    }// END Method existPluginInterface


    @Override
    public void addPluginImplementation(String pluginInterfaceName, String pluginImplementationAddress) {

        plugins.get(pluginInterfaceName).add(pluginImplementationAddress);

    }// END Method addPluginImplementation


    @Override
    public void removePluginImplementation(String pluginInterfaceName, String pluginImplementationAddress) {

        plugins.get(pluginInterfaceName).remove(pluginImplementationAddress);

    }// END Method removePluginImplementation


    @Override
    public boolean existPluginImplementation(String pluginInterfaceName, String pluginImplementationAddress) {

        return existPluginInterface(pluginInterfaceName) && plugins.get(pluginInterfaceName).contains(pluginImplementationAddress);

    }// END Method existPluginImplementation


    @Override
    public Iterable<String> getPluginInterfacesNames() {
        
        return plugins.keySet();

    }// END Method getPluginInterfacesNames


    @Override
    public Iterable<String> getPluginImplementationsAddresses(String pluginInterfaceName) {

        return plugins.get(pluginInterfaceName);

    }// END Method getPluginImplementationsStrings


}// END Class RAMPersistenceDriver
